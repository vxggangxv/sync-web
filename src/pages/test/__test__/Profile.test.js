import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Profile from './Profile';

describe('<Profile />', () => {
  it('matches snapshot', () => {
    const utils = render(<Profile username="juncoder" name="준코더" />);
    expect(utils.container).toMatchSnapshot();
  });
  it('shows the props correctly', () => {
    const utils = render(<Profile username="juncoder" name="준코더" />);
    utils.getByText('juncoder'); // velopert 라는 텍스트를 가진 엘리먼트가 있는지 확인
    utils.getByText('(준코더)'); // (준코더) 이라는 텍스트를 가진 엘리먼트가 있는지 확인
    utils.getByText(/준/i); // 정규식 /준/ 을 통과하는 엘리먼트가 있는지 확인
  });
});
