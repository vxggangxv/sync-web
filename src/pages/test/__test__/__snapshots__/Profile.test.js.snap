// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<div>
  <div>
    <b>
      juncoder
    </b>
     
    <span>
      (
      준코더
      )
    </span>
  </div>
</div>
`;
